src/dropdown/Dropdown.js

```diff
@@ -199,9 +194,25 @@ export default class Dropdown extends React.Component {
   }) => {
 
     const Option = optionComponent;
-    const propertyName = this.props.groupByKey;
+    const { groupByKey } = this.props;
+    const groupedOptions = options.reduce((acc, o, i) => {
+      const shouldCreateNewGroup = i === 0 || o[groupByKey] !== options[i - 1][groupByKey];
+      if (shouldCreateNewGroup) {
+        const newGroup = {
+          optionGroupTitle: o[groupByKey] || '',
```

Comment:
@FrancescoCioria [here](https://github.com/buildo/react-components/blob/9129d45b4bc55fe211fd53a30779259648be4b95/src/dropdown/Dropdown.js#L170-L181) we are checking `groupName !== 'undefined'` as string... it's bad I know :)